pkgs/development/tools/documentation/adrdox/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, dmd }:
+
+stdenv.mkDerivation rec {
+  name = "adrdox-${version}";
+  version = "20190508";
+
+  # adrdox does not have releases or version numbers. We use dates instead,
+  # and map them to commit hashes.
+  commit = {
+    "20190508" = "e253735b1626ddb49d53f3beee9ca1d4b8ac7dda";
+  }.${version};
+
+  src = fetchTarball {
+    url = "https://github.com/adamdruppe/adrdox/archive/${commit}.tar.gz";
+    sha256 = "1cmh7ivxv4kpxgzhd94sahmq6xg9yznh87qnnysqcj546xyxcpd3";
+  };
+  buildInputs = [ dmd ];
+
+  installPhase = ''
+    mkdir --parents $out/bin
+
+    mv doc2 $out/bin
```

Comment:
I believe the install command can be used here which mean the directory does not need to be made first

Reply:
I can't figure out how to do that with install.

It has the -D flag, which seems relevant:

> create all leading components of DEST except the last, or all components of --target-directory, then copy SOURCE to DEST

However, this would only create $out, not $out/bin, because of the "except the last" part.

Nonetheless I made use of install instead of mv now, but still explicitly create the directory.

Reply:
Does it not work if you don't create the directory first? Tbh I am not sure what the install command does but I am sure I have heard that it can create the target directory

Reply:
It seems not to create the directory, I tried that but it said that the directory did not yet exist.

Reply:
Ah I am probably wrong then